src/Inventory/Asset/MainAsset.php

```diff
@@ -645,7 +645,10 @@ public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = 0)
         $entities_id = $this->entities_id;
         $val->is_dynamic = 1;
         $val->entities_id = $entities_id;
-        $val->states_id = $this->states_id_default ?? 0;
+
+        if ($this->states_id_default ?? 0 != '-1') {
```

Comment:
Maybe `0` value should be forced when we are in a new asset creation context, as this `0` value could probably be checked by other rules or plugins in other pieces of code.